content/master/concepts/environment-configs.md

```diff
@@ -0,0 +1,291 @@
+If multiple 
+{{<hover label="comp" line="7">}}environmentConfigs{{</hover>}}
+use the same key, the Composition uses the value of the last environment listed.
+{{</hint >}}
+
+### Select by name
+
+Select an environment by name with
+{{<hover label="byName" line="8">}}type: Reference{{</hover>}}. 
+
+Define the
+{{<hover label="byName" line="9">}}ref{{</hover>}} object and the 
+{{<hover label="byName" line="10">}}name{{</hover>}} matching the exact name of
+the environment.
+
+
+For example, select the 
+{{<hover label="byName" line="7">}}environmentConfig{{</hover>}}
+named
+{{<hover label="byName" line="10">}}example-environment{{</hover>}}
+
+```yaml {label="byName",copy-lines="all"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-composition
+spec:
+  environment:
+    environmentConfigs:
+    - type: Reference
+      ref:
+        name: example-environment
+```
+
+### Select by label
+
+Select an environment by labels with a
+{{<hover label="byLabel" line="8">}}type: Selector{{</hover>}}.  
+
+Define the {{<hover label="byLabel" line="9">}}selector{{</hover>}} object.  
+
+The
+{{<hover label="byLabel" line="10">}}matchLabels{{</hover>}} object contains a
+list of labels to match on. 
+
+Selecting a label requires matching both the label 
+{{<hover label="byLabel" line="11">}}key{{</hover>}} 
+and the value of key. 
+
+When matching the label's value, provide an exact value with a 
+{{<hover label="byLabel" line="12">}}type: Value{{</hover>}} and provide the value
+to match in the 
+{{<hover label="byLabel" line="13">}}value{{</hover>}} field.
+
+Crossplane can also match a label's value based on an input in the composite
+resource. Use 
+{{<hover label="byLabel" line="15">}}type: FromCompositeFieldPath{{</hover>}} 
+and provide the field to match in the 
+{{<hover label="byLabel" line="16">}}valueFromFieldPath{{</hover>}} field.
+
+```yaml {label="byLabel",copy-lines="all"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: example-composition
+spec:
+  environment:
+    environmentConfigs:
+    - type: Selector
+      selector: 
+        matchLabels:
+          - key: my-label-key
+            type: Value
+            value: my-label-value
+          - key: my-label-key
+            type: FromCompositeFieldPath
+            valueFromFieldPath: spec.parameters.deploy
+  resources:
+  # Removed for brevity
+```
+
+#### Manage selector results
+
+Selecting environments by labels may return more than one environment.  
+The Composition sorts all the results by the name of the environments and
+only uses the first environment in the sorted list. 
+
+Set the {{<hover label="selectResults" line="10">}}mode{{</hover>}} as 
+{{<hover label="selectResults" line="10">}}mode: Multiple{{</hover>}} to return
+all matched environments. Use 
+{{<hover label="selectResults" line="19">}}mode: Single{{</hover>}} to 
+return a single environment.
+
+{{<hint "note" >}}
```

Comment:
this hint is realllly helpful to explain the complexity of selecting and dealing with multiple results